tests/Hakyll/Core/Runtime/Tests.hs

```diff
@@ -118,7 +147,73 @@ case03 = do
                 makeItem example
                     >>= loadAndApplyTemplate "partial.html" defaultContext
 
-
     ec @?= ExitFailure 1
 
     cleanTestEnv
+
+
+--------------------------------------------------------------------------------
+-- Test that dependency cycles are correctly identified in the presence of 
+-- snapshots. See issue #878.
+case05 :: Assertion 
+case05 = do
+    logger  <- Logger.new Logger.Error 
+    (ec, _) <- run RunModeNormal testConfiguration logger $ do
+
+        match "posts/*" $ do
+            route $ setExtension "html"
+            compile $ do
+                let applyDefaultTemplate item = do
+                        footer <- loadBody "footer.html"
+                        let postCtx' =
+                                constField "footer" footer `mappend`
+                                defaultContext
+                        loadAndApplyTemplate "template-empty.html" postCtx' item
+
+                pandocCompiler
+                    >>= saveSnapshot "content"
+                    >>= loadAndApplyTemplate "template-empty.html" defaultContext
+                    >>= applyDefaultTemplate
+                    >>= relativizeUrls
+
+        create ["footer.html"] $
+            compile $ do
+                posts <- fmap (take 5) . recentFirst =<< loadAllSnapshots "posts/*" "content"
+                let footerCtx =
+                        listField "posts" defaultContext (return posts) `mappend`
+                        defaultContext
+
+                makeItem ""
+                    >>= loadAndApplyTemplate "template-empty.html" footerCtx
+        
+        create ["template-empty.html"] $ compile templateCompiler
+
+    ec @?= ExitSuccess 
+
+    cleanTestEnv
+
+
+case06 :: Assertion
```

Comment:
This needs a comment, probably with the link to https://github.com/jaspervdj/hakyll/pull/880#discussion_r708650172 that explains what's going on.